### PR TITLE
Implement clean up routine

### PR DESCRIPTION
**Pull request by Mewbi**
The deploy of this PR requires change database, it can be made just deleting the old database file or adding manually the new required column with:

```sql
ALTER TABLE retrospectives ADD COLUMN created_at DATETIME DEFAULT CURRENT_TIMESTAMP;
```

### CODE REVIEW

File: internal/repository/websocket.go [hunk 1]
```diff
@@ -162,6 +162,10 @@ func (w *WebSocket) DeleteQuestion(ctx context.Context, id uuid.UUID) (*types.Qu
 	return nil, w.sendMessageToRetro(ctx, message, nil)
 }
 
+func (s *WebSocket) GetOldRetrospectives(ctx context.Context, date time.Time) ([]uuid.UUID, error) {
+	panic("unimplemented")
```
Comment (ySnoopyDogy) [inline internal/repository/websocket.go:166]: Does not CleanUpRetros need this to be implemented to work??

**Review by ySnoopyDogy — DISMISSED** [review]

File: internal/repository/websocket.go (hunk 1 above)
Comment (Mewbi) [reply internal/repository/websocket.go:166]: That's just the websocket interface, actually don't matter to websocket this function, but is required because websocket must implement the repository interface

File: internal/service/service.go [hunk 2]
```diff
@@ -36,14 +40,22 @@ func (s *Service) CreateRetrospective(ctx context.Context, retro *types.Retrospe
...
 }
 
 func (s *Service) DeleteRetrospective(ctx context.Context, id uuid.UUID) (*types.Retrospective, error) {
+	config := config.Get()
+	cleanUpDays := time.Duration(config.Schedule.CleanUpDays)
 	retro, err := s.repository.DeleteRetrospective(ctx, id)
 	if err != nil {
 		return nil, err
 	}
+	retro.ExpireAt = retro.CreatedAt.Add(cleanUpDays * 24 * time.Hour)
```
Comment (Termack) [inline internal/service/service.go:58]: Why is this here?

File: internal/service/service.go [hunk 3]
```diff
@@ -36,14 +40,22 @@ func (s *Service) CreateRetrospective(ctx context.Context, retro *types.Retrospe
 }
 
 func (s *Service) GetRetrospective(ctx context.Context, id uuid.UUID) (*types.Retrospective, error) {
-	return s.repository.GetRetrospective(ctx, id)
+	config := config.Get()
+	cleanUpDays := time.Duration(config.Schedule.CleanUpDays)
+
+	retro, err := s.repository.GetRetrospective(ctx, id)
+	retro.ExpireAt = retro.CreatedAt.Add(cleanUpDays * 24 * time.Hour)
```
Comment (Termack) [inline internal/service/service.go:47]: Why is this here?

File: internal/server/schedule.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package server
+
+import (
+	"api/config"
+	"context"
+	"log"
+	"time"
+)
+
+func (c *controller) StartSchedule() {
```
Comment (Termack) [inline internal/server/schedule.go]: I think it would be better if this logic was in main or inside controller.Start

File: internal/service/service.go [hunk 5]
```diff
@@ -137,3 +149,22 @@ func (s *Service) LoadAllRetrospectives(ctx context.Context) error {
...
+		return err
+	}
+
+	for _, id := range ids {
+		if _, err := s.repository.DeleteRetrospective(ctx, id); err != nil {
+			return err
+		}
+	}
+
+	log.Printf("deleted %d retrospectives older than %s", len(ids), date.String())
```
Comment (Termack) [inline internal/service/service.go:168]: maybe add a \n in the end of this print

**Review by Termack — CHANGES_REQUESTED** [review]

File: internal/service/service.go (hunk 2 above)
Comment (Mewbi) [reply internal/service/service.go:58]: It can be used to frontend eventually to show when the retro will expire.

As cleanUpDays is a config value, it easier to backend send this.

Comment (Mewbi) [reply internal/service/service.go:58]: Obs. Technically this information is not necessary in this function, I just insert this because it returns the struct retro.

File: internal/service/service.go (hunk 3 above)
Comment (Mewbi) [reply internal/service/service.go:47]: ditto

File: internal/service/service.go (hunk 5 above)
Comment (Mewbi) [reply internal/service/service.go:168]: Not necessary, this function already break line after print

**Review by Termack — APPROVED** [review]
